src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1356,6 +1361,85 @@ testUnsafeSendChan TestTransport{..} = do
 
   takeMVar clientDone
 
+testCallLocal :: TestTransport -> Assertion
+testCallLocal TestTransport{..} = do
+  node <- newLocalNode testTransport initRemoteTable
+
+  -- Testing that (/=) <$> getSelfPid <*> callLocal getSelfPid.
+  result <- newEmptyMVar
+  runProcess node $ do
+    r <- (/=) <$> getSelfPid <*> callLocal getSelfPid
+    liftIO $ putMVar result r
+  True <- takeMVar result
+  return ()
+
+  -- Testing that when callLocal is interrupted, the worker is interrupted.
+  ibox <- newIORef False
+  runProcess node $ do
+    keeper <- getSelfPid
+    spawnLocal $ do
+        caller <- getSelfPid
+        onException 
+          (callLocal $ do
+                onException (do send keeper caller
+                                expect)
+                            (do liftIO $ writeIORef ibox True))
+          (send keeper ())
+    caller <- expect
+    exit caller "test"
+    ()     <- expect
+    return ()
+  True <- readIORef ibox
+  return ()
+    
+  -- Testing that when the worker raises an exception, the exception is propagated to the parent.
+  ibox2 <- newIORef False
+  runProcess node $ do
+    r <- try (callLocal $ error "e" >> return ())
+    liftIO $ writeIORef ibox2 (r == Left (ErrorCall "e"))
+  True <- readIORef ibox
+  return ()
+
+  -- Test that worker waits for the caller in correct situation
+  ibox3 <- newIORef False
+  result3 <- newEmptyMVar
+  runProcess node $ do
+    keeper <- getSelfPid
+    spawnLocal $ do
+        callLocal $ 
+            (do us <- getSelfPid
+                send keeper us
+                () <- expect
+                liftIO yield)
+            `finally` (liftIO $ writeIORef ibox3 True)
+        liftIO $ putMVar result3 =<< readIORef ibox3
+    worker <- expect
+    send worker ()
+  True <- takeMVar result3
+  return ()
+
+  -- Test that worker waits for the caller in case of exception in parent
```

Comment:
Maybe **... when caller gets an exception.** instead of **parent**, if **caller** and **parent** refer to the same thing.